3-extra/2-array-of-objects.js

```diff
@@ -12,6 +12,14 @@
 
 function getHighestRatedInEachGenre(books) {
     // TODO
+    let bookTitles=[];
+    for(let i=0; i<books.length; i++){
+        if(books[i].rating > 4.8){
+         bookTitles.push(books[i].title)   
+        }
+        
+    }
+return bookTitles;
```

Comment:
👍 It's great that you've solved the problem by adding the highest rating manually. I am so happy you started looking into the extra exercises too.

For finding the highest rating with the code (so that you can use the function with other types of books array and rating that `4.8` is not their highest rating), primary a for loop over books before what you wrote, can find the max rating first, then you can replace it with the `4.8` in your above code. For that, you can use a variable with the max rating of `0` (minimum rating) and then if you find a book with more ratings than that, you update the maximum rating till you visit all the book ratings.